pennylane/interfaces/tf.py

```diff
@@ -174,6 +179,9 @@ def _evaluate_grad_matrix(grad_matrix_fn):
             if grad_matrix_fn in saved_grad_matrices:
                 return saved_grad_matrices[grad_matrix_fn]
 
+            if self.jacobian_options.get("cache_state", False):
+                self.jacobian_options["device_pd_options"] = {"starting_state": state}

```

Comment:
Could we just have this line within the `if jac_options.get("cache_state", False):` statement above? Is there a reason to do it within `_evaluate_grad_matrix`?

Reply:
`_evaluate_grad_matrix` could be called later on after arbitrary other executions on the device.  This function `_evaluate_grad_matrix` will remember the value of state where the function was constructed.  It basically caries around the state with it.

I don't want to set the tape keyword till the last moment before taking the derivative in case it gets overwritten by a different execution.

Reply:
This keyword gets set during qnode initialization time.  Directly manipulating tape jacobian options isn't particularly user-facing, but I don't think an overabundance of caution here will hurt anyone.